README.md

```diff
@@ -0,0 +1,110 @@
+```
+
+Configuration of the gateway is done similar to the Docker container, using the same config options as listed [here](https://github.com/xoseperez/basicstation-docker?tab=readme-ov-file#configure-the-gateway).
+The only difference is that we use snap configs rather than environment variables.
+For the variable name (key), change it to lower case and replace underscores with hyphens.
+Then prefix the key with `env.`.
+The value remains exactly the same as before.
+
+Example:
+`GATEWAY_EUI=DEADFFFEBEEF` becomes `env.gateway-eui=DEADFFFEBEEF`
+
+After you have the configs in this format, you can set them one by one using the `snap set` command:
+
+```
+sudo snap set lora-basicstation env.tc-uri="wss://eu1.cloud.thethings.network:8887"
+sudo snap set lora-basicstation env.tc-key=NNSXS...
+```
+
+Or with a single command:
+
+```
+sudo snap set lora-basicstation env.gateway-eui=DEADFFFEBEEF env.model=SX1301 \
+env.tc-key=NNSXS... env.tc-uri=wss://eu1.cloud.thethings.network:8887 \
+env.device=/dev/spidev0.0 env.interface=SPI env.spi-speed=200000 \
+env.use-libgpiod=1 env.gpio-chip=gpiochip4 env.reset-gpio=22
+```
+
+The gateway software starts automatically after it is installed.
+You can however manually start it with this command:
+
+```
+sudo snap start lora-basicstation
+```
+
+To view the log output, run:
+
+```
+sudo snap logs -f lora-basicstation
+```
+
+The gateway software can be stopped using:
+
+```
+sudo snap stop lora-basicstation
+```
+
+## Tested platforms
+
+- Ubuntu Server 24.04 + Raspberry Pi 5 + [Pi Supply IoT Gateway Hat](https://uk.pi-supply.com/products/iot-lora-gateway-hat-for-raspberry-pi) + [RAK833-SPI](https://www.rakwireless.com/en-us/products/lpwan-gateways-and-concentrators/rak833)
+- Ubuntu Core 24 + Raspberry Pi 5 + [Pi Supply IoT Gateway Hat](https://uk.pi-supply.com/products/iot-lora-gateway-hat-for-raspberry-pi) + [RAK833-SPI](https://www.rakwireless.com/en-us/products/lpwan-gateways-and-concentrators/rak833)
+
+## Debugging
+
+### Verify Docker container is running or not
+
+To make sure the docker container is running, execute the following command.
+If it does not list any containers with the name `lora-basicstation`, the container is not running.
+
+```
+sudo docker ps
+```
+
+In edge cases during development there might be a `lora-basicstation` docker container running that is not controlled by the companion snap.
+You can manually stop it using:
+
+```
+sudo docker stop lora-basicstation
+sudo docker rm -f lora-basicstation
+```
+
+### Reset concentrator
+
+To manually reset the gateway concentrator hardware, you can toggle the reset GPIO pin. A conveniance script is provided to assist with this. The following example is for the Raspberry Pi 5. Replace `RESET_GPIO="22"` with the correct GPIO your concentrator is connected to. If you are using an older Raspberry Pi, set `GPIO_CHIP` to `gpiochip0`.
+
+```
+sudo GPIO_CHIP="gpiochip4" RESET_GPIO="22" POWER_EN_GPIO=0 POWER_EN_LOGIC=0 ./reset.gpiod.sh
```

Comment:
Can this be packages in the same snap?

Reply:
This reset script, along with a couple other are included in the docker container. We could expose them as separate apps in the snap. I'd rather leave that for a future task.